### PR TITLE
Support raw Pods / No URLs

### PR DESCRIPTION
**Pull request by jerrymarino**
It is desireable to integrate pods that are distributed as a raw package. Such
Pods don't have a URL, since they aren't hosted.

The proposed approach is simple:

- A user copies the pod to Vendor/Pod.
- No `url` is provided.
- update_pods.py will run RepoTool in that directory.

This allows a user to add in arbitrary source code, which there is really no way
around in this case.

### CODE REVIEW

File: bin/update_pods.py [hunk 1]
```diff
@@ -157,6 +157,24 @@ def _needs_update(repository_ctx):
...
+        # We allow putting source code in the Vendor/PodName and then initing
+        # the repo with that code.
+	return
+
+    # Note: the pod is not cleaned out if the sourcecode is loaded from the
+    # current directory
+    _exec(repository_ctx, ["rm", "-rf", repository_ctx.GetPodRootDir()])
+    _exec(repository_ctx, ["mkdir", "-p", repository_ctx.GetPodRootDir()])
+
+    if url.startswith("http") or url.startswith("https"):
```
Comment (rahul-malik) [inline bin/update_pods.py]: Nit: don't you have to only check "http"?

File: bin/update_pods.py [hunk 2]
```diff
@@ -157,6 +157,24 @@ def _needs_update(repository_ctx):
...
+        _fetch_remote_repo(repository_ctx, repo_tool_bin_path, target_name, url)
+    elif url.startswith("/"):
+        _link_local_repo(repository_ctx, target_name, url)
```
Comment (rahul-malik) [inline bin/update_pods.py]: There might be better APIs to determine if a url is file or remote

File: bin/update_pods.py [hunk 3]
```diff
@@ -377,6 +398,11 @@ def main():
 
     global SRC_ROOT
     SRC_ROOT = args.src_root
+
+    if not SRC_ROOT:
+        SRC_ROOT = os.getcwd()
```
Comment (rahul-malik) [inline bin/update_pods.py]: Might want to print a warning here that src_root was not passed in and we are falling back to cwd

File: bin/update_pods.py (hunk 1 above)
Comment (jerrymarino) [reply bin/update_pods.py]: @rahul-malik you are right!

File: bin/update_pods.py (hunk 2 above)
Comment (jerrymarino) [reply bin/update_pods.py]: I'm not familiar with such an API but that would bee pretty sweet. The nuance here is we either support absolute paths ( of for temporary local dev ) or remote URL's.

File: bin/update_pods.py (hunk 3 above)
Comment (jerrymarino) [reply bin/update_pods.py]: @rahul-malik this is a good idea - we should deff print the `SRC_ROOT` somewhere in output.

Comment (jerrymarino) [reply bin/update_pods.py]: It looks like we already print this out:
```
405     print("Updating pods in " + SRC_ROOT)
```

**Pull request merged**